src/main/kotlin/no/nav/faktureringskomponenten/service/beregning/AntallBeregner.kt

```diff
@@ -0,0 +1,64 @@
+package no.nav.faktureringskomponenten.service.beregning
+
+import mu.KotlinLogging
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+
+private val log = KotlinLogging.logger { }
+
+class AntallBeregner {
+    companion object {
+        fun antallForPeriode(fom: LocalDate, tom: LocalDate): BigDecimal {
+            val gjelderSammeMånedOgÅr = fom.year == tom.year && fom.monthValue == tom.monthValue
+            if (gjelderSammeMånedOgÅr) {
+                log.debug("Beregner antall for periode $fom og $tom på én og samme måned")
+                return beregnAntallForEnMåned(fom, tom)
+            }
+            log.debug("Beregner antall for periode $fom og $tom på flere måneder")
+            return beregnAntallForFlereMåneder(fom, tom)
```

Comment:
Dette blir feil, hvis man skal beregne etter det jeg tolker er riktig beregning. Her beregner du altså altså en prosentandel på antall dager i periode / antall dager i alle måneder, noe som gir deg en total prosentandel periodeDager av alleDagerIMåneden.

Riktige beregning etter min tolkning:
La oss ta eksempel 14. februar 2023 til 13. april 2023, med prisPerMåned=10000.
Februar: 14/30=0.5 -> 10000*0.5 = 5000
Mars: 31/31 = 1 -> 10000*1 = 10000
April: 13/30 -> 0.4333 -> 4330

Dermed blir det 5000+10000+4330.

Reply:
Hvis det er noe feil bør vi ta det opp i felleskap for å sjekke regnestykket ytterligere :+1:

Reply:
Neida, som jeg skrev i EDIT-versjonen av posten så mistolket jeg koden din. Din løsning er samme løsningen som jeg kom med selv, men jeg er fortsatt ikke er enig med skaleringen. Men det er ikke min plass å bestemme, jeg kan bare kommentere.

Reply:
Hele grunnen til at vi valgte BigDecimal i utgangspunktet var for å være nøytaktig i beregningene våre.